firmware/ventilator-controller-stm32/Core/Src/Pufferfish/Driver/Power/Simulator.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Simulator.cpp
+ *
+ *  Created on: May 30, 2021
+ *      Author: Rohan
+ *
+ *  Simulator for power management
+ */
+
+#include "Pufferfish/Driver/Power/Simulator.h"
+
+#include <cmath>
+#include <random>
+
+#include "Pufferfish/Util/Timeouts.h"
+
+namespace Pufferfish::Driver::Power {
+
+// NOLINTNEXTLINE(cert-msc51-cpp)
+std::ranlux24_base prng;
+// NOLINTNEXTLINE(readability-magic-numbers)
+std::uniform_int_distribution<int> uniform_int(1, 100);
+static constexpr float uniform_width = 1;
+std::uniform_real_distribution<float> uniform_centered(-uniform_width / 2, uniform_width / 2);
+
+void Simulator::input_clock(uint32_t current_time) {
+  if (initial_time_ == 0) {
```

Comment:
The `initial_time` related code in the simulator backend was only so that the simulator backend's timestamps would resemble the firmware's timestamps, as Python's `time.time()` has time=0 at Jan 1 1970 while the firmware's `HAL::time()` has time=0 when the firmware first started. We should remove `initial_time_` from the firmware's power simulator.

Reply:
Actually, the Breathing Circuit simulator also has `initial_time` (though maybe it shouldn't either). We can leave `initial_time` in the power simulator for this PR, just to be consistent.